lib/class-wp-theme-json-gutenberg.php

```diff
@@ -2230,6 +2259,49 @@ public function get_styles_for_block( $block_metadata ) {
 			}
 		}
 
+		// If there are style variations, generate the declarations for them, including any feature selectors the block may have.
```

Comment:
Not a blocker, but I wonder if in a follow-up there's an opportunity to either consolidate some of this logic with the feature declarations above, or split it out into a separate function, since this function is now quite long? Definitely not something to worry about in this PR, just jotting down the thought 🙂